### ISSUE TITLE
How to calculate the loss of multi-turn dialogue training data?

### ISSUE THREAD

**Issue by NUMB1234** [huggingface/trl#2424]
In a single data entry containing multiple turns of dialogue, abbreviated as Q1 + A1 + Q2 + A2, does this project calculate the loss only for the last answer of the multi-turn dialogue, or for each answer?

**Comment by qgallouedec** [comment]
Which method does your question refer to?

**Comment by qgallouedec** [comment]
By default the loss is computed over the whole sequence. 

You can make sft train ignore some parts of the sequence by using a custom data collator (see sft doc)

**Comment by NUMB1234** [comment]
I'm sorry, perhaps I didn't clarify my question. For multi-turn dialogue data in this format: 
{"messages": [{"role": "system", "content": "..."}, {"role": "user", "content": "..."}, {"role": "assistant", "content": "..."}, {"role": "user", "content": "..."}, {"role": "assistant", "content": "..."}, ...]} 
Does this project calculate the loss only for the last assistant's response, or calculate the loss for all assistant responses? I couldn't find an explanation in the documentation.

**Comment by twotwoiscute** [comment]
> Neither. By default, the loss is computed on the full sequence, ie all assistant and user messages.

Hi, I would like to know where I can view this part of code , Thanks.
